### PR TITLE
Fix all latex expressions without enclosing $ signs

### PR DESCRIPTION
**Pull request by rheaparekh**
Fix all latex expressions without the enclosing $ signs. Fixes issue #3286

Approach: I searched for all the raw strings in the repository, and went these strings one by one to check for missing latex expressions.

Some comments regarding specific changes to make sure they are correct:

* For file `cirq/linalg/transformations.py`: The changes would look like [this](https://hackmd.io/0UK6Na6mQ1aY2tkcEFSlxg?view).

* For file `examples/bcs_mean_field.py`: The changes would look like [this](https://hackmd.io/ZpqO-TdnQbajFo_TSRBiCA?view). I have changed the command `\dag` to `\dagger` as `\dag` wasn't being rendered properly in HackMd.

* For file `cirq/ops/common_channels.py`: Changed `$ \sum_i p_i Pi \rho Pi $` to  `$\sum_i p_i Pi \rho Pi$`, as the first expression isn't being rendered correctly in HackMd ([link](https://hackmd.io/v8_W6pXjTsqj2KuSXlJK6w?view) to view this).

### CODE REVIEW

File: examples/bcs_mean_field.py [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 The Bogoliubov transformation can be readily implemented by
 applying quantum gates on vertical pairs of qubits, which takes the form
-|BCS⟩ = \prod_k (u_k + v_k c^\dag_{k,↑} c^\dag_{−k,↓}|vac⟩ where |vac⟩ is
-the vacuum state and u_k^2 = (1+ ξ_k/(ξ_k^2+Δ_k^2)^{1/2})/2 and v_k^2
-= (1 - ξ_k/(ξ_k^2+Δ_k^2)^{1/2})/2.
+$|BCS⟩ = \prod_k (u_k + v_k c^\dagger_{k,↑} c^\dagger_{−k,↓}|vac⟩$ where $|vac⟩$
```
Comment (balopat) [inline examples/bcs_mean_field.py:13]: Missing parenthesis: 

```suggestion
$|BCS⟩ = \prod_k (u_k + v_k c^\dagger_{k,↑} c^\dagger_{−k,↓}|vac⟩)$ where $|vac⟩$
```

File: cirq/experiments/qubit_characterizations.py [hunk 2]
```diff
@@ -411,45 +411,46 @@ def two_qubit_state_tomography(sampler: work.Sampler,
...
     is applied to the second qubit before measurement, the measurement
-    operators are (I -/+ sigma_x) \bigotimes (I +/- sigma_y). The probabilities
-    obtained instead contribute to the following linear equations:
+    operators are $(I -/+ \sigma_x) \bigotimes (I +/- \sigma_y)$. The
+    probabilities obtained instead contribute to the following linear equations:
 
-    c_02 - c_10 - c_12 = 4*P_00 - 1
-    -c_02 - c_10 + c_12 = 4*P_01 - 1
-    c_02 + c_10 + c_12 = 4*P_10 - 1
+    $c_{02} - c_{10} - c_{12} = 4*P_{00} - 1$
```
Comment (balopat) [inline cirq/experiments/qubit_characterizations.py:439]: Let's use $$ here too

**Review by balopat — CHANGES_REQUESTED** [review]
Thank you for combing through this! I added some comments to fix.

**Comment by rheaparekh** [comment]
Thank you @balopat for the review. I have made the requested changes.

I also observed that I had missed the file `cirq/ops/parity_gates.py`, and I added new changes in that file.

**Review by balopat — APPROVED** [review]

File: cirq/ops/parity_gates.py [hunk 3]
```diff
@@ -121,20 +133,32 @@ def __repr__(self) -> str:
 class YYPowGate(eigen_gate.EigenGate,
                 gate_features.TwoQubitGate,
                 gate_features.InterchangeableQubitsGate):
-    """The Y-parity gate, possibly raised to a power.
+    r"""The Y-parity gate, possibly raised to a power.
```
Comment (balopat) [inline cirq/ops/parity_gates.py:136]: why the r? 
```suggestion
    """The Y-parity gate, possibly raised to a power.
```

File: cirq/ops/parity_gates.py [hunk 4]
```diff
@@ -30,20 +30,32 @@
 class XXPowGate(eigen_gate.EigenGate,
                 gate_features.TwoQubitGate,
                 gate_features.InterchangeableQubitsGate):
-    """The X-parity gate, possibly raised to a power.
+    r"""The X-parity gate, possibly raised to a power.
```
Comment (balopat) [inline cirq/ops/parity_gates.py:33]: Why the r? 
```suggestion
    """The X-parity gate, possibly raised to a power.
```

Comment (balopat) [reply cirq/ops/parity_gates.py:33]: Ignore me - this is fine. raw strings are best for backslash to be interpreted as a character.

**Comment by rheaparekh** [comment]
Thank you for the review. I am glad to have my first PR merged here!

**Pull request merged**